src/common/commonutils/FileUtils.c

```diff
@@ -81,7 +81,7 @@ static bool SaveToFile(const char* fileName, const char* mode, const char* paylo
             {
                 for (i = 0; i < payloadSizeBytes; i++)
                 {
-                    if (payload[i] != fputc(payload[i], file))
+                    if (payload[i] != (char)fputc(payload[i], file))

```

Comment:
If payload[i] == -1 and fputc fails, what's the result of the comparison?

Reply:
fputc internally converts to unsigned char, and then return that as a signed integer. Then if payloiad[i] is a negative char, comparison fails and there is a logged trace saying why. For this function, payload does not have to be unsigned chars.